lib/link_expansion/edition_hash.rb

```diff
@@ -0,0 +1,36 @@
+class LinkExpansion::EditionHash
+  class << self
+    def from(values)
+      return nil unless values.present?
+      hash = values.is_a?(Array) ? hash_for(values) : values
+      hash = SymbolizeJSON.symbolize(hash)
+      hash = hash.slice(*edition_fields)
+      hash[:api_path] = api_path(hash) unless hash[:base_path].nil?
+      hash[:withdrawn] = withdrawn?(hash)
+      hash.except(:id, :"unpublishings.type")
+    end
+
+    def edition_fields
+      LinkExpansion::Rules::DEFAULT_FIELDS_WITH_DETAILS.dup +
```

Comment:
I don't think we need the `.dup` here since that'll happen implicitly when using `+` and `-`.